raiden-dapp/src/service-worker/assistant.ts

```diff
@@ -25,4 +44,9 @@ export default class ServiceWorkerAssistant {
       console.warn(`Failed to get (a valid) version: ${error.message}`); // eslint-disable-line no-console
     }
   };
+
+  private reloadWindow = (): void => {
+    // Delay reload so that the service worker can unregister safely if necessary.
+    setTimeout(() => window.location.reload(), 1000);
```

Comment:
Maybe mention why you chose a one second delay here.

Reply:
Okay. So I would write it here and you can argue if that make sense to you:
1s should be very long from the execution time as the unregistration of the service worker is a direct follow up. In terms anything else is taking the CPU 1s should be still enough so the server worker gets the CPU again to unsubscribe. But 1s is short enough that the user feels not like the app is laggy. As this is an update he should  be fine that it is not instant. 

Now thinking about it we could improve it and wait until no service worker is controlling the client anymore (like for the check if there is already a controller during registration). That would make it bullet-proof. 

What do you think?

Reply:
I'm implementing the smarter and safe solution now and test it. Thanks for pushing me for a better solution!

Reply:
Even better. I did some wrong assumptions. Unregistering a service worker makes it only switching into the redundant state. But as long as there is now new service worker in the pipe that got installed and can be activated, the redundant worker is still the controlling one. Means we can still exchange messages with the client(s). Therefore we can savly first unregister the service worker itself and then notify the client to reload the window. To be 100% sure I tested this with an timeout of 20s between the instructions of the service worker to unregister and sending the message. Finally the assistant must not wait for anything with a wild guess timeout or check any conditions. Just reload the window right away.
Checkout the commit for details (even i think this comment might be even longer...)

Reply:
I'm closing this optimistically as I think I have successfully targeted your concerns in the best way possible.